test/puppetlabs/trapperkeeper/services/status/status_core_test.clj

```diff
@@ -15,4 +15,17 @@
 
       (is (nil? (schema/check ServicesInfo @status-fns)))
       (is (= 2 (count (get @status-fns "foo"))))
-      (is (= 1 (count (get @status-fns "bar")))))))
+      (is (= 1 (count (get @status-fns "bar")))))
+
+    (testing (str "registering a service status callback function with a "
+                  "version that already exists causes an error")
+      (is (thrown? IllegalStateException
+                   (update-status-context status-fns "foo"
+                                          "1.1.0" 2 (fn [] "foo repeat")))))
+
+    (testing (str "registering a service status callback function with a "
+                  "status version that already exists but a different service "
+                  "version causes an error")
+      (is (thrown? IllegalStateException
+                   (update-status-context status-fns "foo"
+                                          "1.2.0" 2 (fn [] "foo repeat")))))))
```

Comment:
We definitely want to support more than one `status-version`, totally agree.  But `service-version`?  That'd be more like if we registered callbacks for `puppet-server` version `2.4.9` and `2.5.0` in the same process... which seems... craycray?  I might be missing something but so far I haven't been able to imagine a situation where that seems like it would make sense.